obspy/io/rg16/core.py

```diff
@@ -70,6 +70,11 @@ def _read_rg16(filename, headonly=False, starttime=None, endtime=None,
             trace = _make_trace(filename, trace_block_start, headonly,
                                 contacts_north, details)
             traces.append(trace)
+        if trace_starttime <= starttime.timestamp < trace_endtime and\
+           trace_starttime < endtime.timestamp <= trace_endtime:
+            trace = _make_trace(filename, trace_block_start, headonly,
+                                contacts_north, details)
+            traces.append(trace)
```

Comment:
Maybe it would be more clear to just check if `starttime` or `endtime` exclude the current block, else include it? I am thinking something like this:

```python
    for i in range(0, record_count):
        nbr_bytes_trace_block = _cmp_jump(filename, trace_block_start)
        trace_starttime = _read(filename, trace_block_start + 20 + 2 * 32, 8,
                                'binary') / 1e6
        trace_endtime = trace_starttime + record_length
        # determine if this block is excluded based on start/endtime params
        con1 = starttime.timestamp > trace_endtime
        con2 = endtime.timestamp < trace_starttime
        if con1 or con2:  # if so increment trace_block_start and continue
            trace_block_start += nbr_bytes_trace_block
            continue
        # else collect trace 
        trace = _make_trace(filename, trace_block_start, headonly,
                            contacts_north, details)
        traces.append(trace)
        trace_block_start += nbr_bytes_trace_block
```

Reply:
Thank you for your advices Chambers, i took into account them in the new commit.